event-based-simulators/extras/GenerateMeasurements.py

```diff
@@ -0,0 +1,248 @@
+import sys
+import json, os, logging, requests, base64
+from datetime import datetime, timedelta, time
+from random import randint, uniform, choices, gauss
+from statistics import mean
+
+from c8y_api import CumulocityApi
+
+import ArgumentsAndCredentialsHandler, Environment
+
+C8Y_PROFILE_GROUP = 'c8y_EventBasedSimulatorProfile'
+C8Y_OEE_SIMULATOR_DEVICES_GROUP = "c8y_EventBasedSimulator"
+c8y = CumulocityApi(base_url=Environment.C8Y_BASE,  # the url of your Cumulocity tenant here
+                    tenant_id=Environment.C8Y_TENANT,  # the tenant ID of your Cumulocity tenant here
+                    username=Environment.C8Y_USER,  # your Cumulocity IoT username
+                    password=Environment.C8Y_PASSWORD)  # your Cumulocity IoT password
+C8Y_HEADERS, MEASUREMENTS_HEADERS = ArgumentsAndCredentialsHandler.SetupHeadersForAPIRequest(
+    tenant_id=c8y.tenant_id, username=c8y.username, password=Environment.C8Y_PASSWORD)
+
+
+def current_timestamp(format="%Y-%m-%dT%H:%M:%S.%f"):
+    return datetime.utcnow().strftime(format)[:-3] + 'Z'
+
+
+logging.basicConfig(format='%(asctime)s %(name)s:%(message)s', level=logging.DEBUG)
+log = logging.getLogger("Measurements-generator")
+log.info(f"started at {current_timestamp()}")
+#####################################################
+'''
+# Check if connection to tenant can be created
+tenantConnectionResponse = ArgumentsAndCredentialsHandler.CheckTenantConnection(baseUrl=Environment.C8Y_BASE,
+                                                                                C8Y_HEADERS=C8Y_HEADERS)
+if tenantConnectionResponse:
+    log.info(f"Connect to tenant {Environment.C8Y_TENANT} successfully")
+else:
+    if tenantConnectionResponse is None:
+        log.error(f"Wrong base url setup. Check again the URL: {Environment.C8Y_BASE}")
+    else:
+        log.error(tenantConnectionResponse.json())
+    log.error(f"Connect to tenant {Environment.C8Y_TENANT} failed")
+    sys.exit()
+'''
+######################################################
+# Start and end date are mandatory and need to be in ISO8601 format (e.g. 2020-10-26T10:00:00.000).
+sim_time = datetime.utcnow()
+sim_end_time = sim_time + timedelta(hours=1)
+
+LIST_OF_MEASUREMENTS = ['Availability', 'AvailabilityLossTime', 'PerformanceLossTime', 'ActualProductionTime',
+                        'QualityLossAmount', 'IdealQualityTime', 'IdealCycleAmount', 'AvailabilityLossAmount',
+                        'IdealMachineRuntime', 'IdealProductionAmount', 'OEE', 'ActualQualityAmount',
+                        'ActualProductionAmount', 'IdealAmount', 'PerformanceLossAmount', 'Quality', 'IdealCycleTime',
+                        'QualityLossTime', 'Performance', 'PotentialProductionTime']
+
+
+def ListAllChildDevices():
+    deviceInTenantCount = 0
```

Comment:
Isnt this just len(deviceManagedObject)

Reply:
deviceManagedObject is an object so it can not be counted by len(). It must be iterated by for loop to be read.

Reply:
I would think there must be a way, if it is iterable. E.G.:
sum(1 for e in deviceManagedObject) 
or
len(list(deviceManagedObject))